embrace-android-core/src/main/kotlin/io/embrace/android/embracesdk/internal/capture/session/SessionPropertiesServiceImpl.kt

```diff
@@ -35,19 +40,30 @@ public class EmbraceSessionPropertiesService(
         }
         val sanitizedKey = enforceLength(originalKey, SESSION_PROPERTY_KEY_LIMIT)
 
-        val removed = sessionProperties.remove(sanitizedKey)
+        val removed = props.remove(sanitizedKey)
         if (removed) {
-            dataSourceProvider()?.apply {
-                removeProperty(sanitizedKey)
-            }
-            listener(sessionProperties.get())
+            dataSourceProvider()?.removeProperty(sanitizedKey)
+            listener?.invoke(props.get())
         }
         return removed
     }
 
-    override fun getProperties(): Map<String, String> = sessionProperties.get()
+    override fun getProperties(): Map<String, String> = props.get()
+
+    override fun populateCurrentSession(): Boolean {
+        return dataSourceProvider()?.addProperties(getProperties()) ?: false
+    }
 
-    override fun populateCurrentSession(): Boolean = dataSourceProvider()?.addProperties(getProperties()) ?: false
+    override fun clearTemporary() {
+        val removedProps = props.clearTemporary()
+        removedProps.forEach { (key, _) ->
```

Comment:
I don't think this is necessary anymore. Now that we've moved the population to after the temp clearing in the service happens, the new session span have the temporary props that were there before, so clearing the service will be sufficient